### ISSUE TITLE
Make generation of random crops fast

### ISSUE THREAD

**Issue by soumik12345** [soumik12345/restorers#17]
The current implementation of our [random crop](https://github.com/soumik12345/mirnetv2/blob/main/mirnetv2/dataloader/base/base_dataloader.py#L37) function uses slicing. We should be able to make a TensorFlow implementation of this operation to make it faster.

**Comment by ariG23498** [comment]
From slack:
```python
concatenated_image = tf.concat([input_image, gt_image], axis=-1)
cropped_concatenated_image = tf.image.random_crop(concatenated_image, (256, 256, 6))
cropped_input_image, cropped_gt_image = tf.split(
    cropped_concatenated_image, num_or_size_splits=2, axis=-1
)
```
This means the entire batch will have the same random crop. Do we want this behaviour?
If yes, then we can use this easily.

**Comment by soumik12345** [comment]
Given that we are training with a pretty low latch size, it probably would not have any adverse effect on the training. We should perform some experiments to investigate this. I think that we should brainstorm and investigate further approaches to tackle this issue.

**Comment by ariG23498** [comment]
I will link this to the PR that I am developing!

**Comment by SauravMaheshkar** [comment]
Closed by #20
